### PR TITLE
add `hasChatAgent` context key for terminal

### PR DESCRIPTION
**Pull request by meganrogge**
terminal part of #215526

### CODE REVIEW

File: src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatActions.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ registerActiveXtermAction({
 	precondition: ContextKeyExpr.and(
 		ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated),
 		// TODO: This needs to change to check for a terminal location capable agent
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatActions.ts:32]: Can be removed?

File: src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.chat.contribution.ts [hunk 2]
```diff
@@ -11,6 +11,9 @@ import { TerminalChatController } from 'vs/workbench/contrib/terminalContrib/cha
 
 registerTerminalContribution(TerminalChatController.ID, TerminalChatController, false);
 
+registerWorkbenchContribution2(TerminalChatEnabler.Id, TerminalChatEnabler, WorkbenchPhase.AfterRestored);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.chat.contribution.ts:14]: This is a workbench contribution so it should be in the "Contributions" section below

**Review by Tyriar — DISMISSED** [review]

**Comment by Tyriar** [comment]
Comments don't seem to be resolved, probably forgot to push

File: src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.chat.contribution.ts [hunk 3]
```diff
@@ -18,12 +18,16 @@ registerTerminalContribution(TerminalChatController.ID, TerminalChatController,
...
+
 // #endregion
 
 // #region Actions
 
 import 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChatActions';
 import { AccessibleViewRegistry } from 'vs/platform/accessibility/browser/accessibleViewRegistry';
 import { TerminalChatAccessibilityHelp } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChatAccessibilityHelp';
+import { registerWorkbenchContribution2, WorkbenchPhase } from 'vs/workbench/common/contributions';
+import { TerminalChatEnabler } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChatEnabler';
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/terminalContrib/chat/browser/terminal.chat.contribution.ts:31]: nit, I think that these imports should be above their usage

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**